### DISCUSSION TITLE
Q1 Summary blog post

### DISCUSSION THREAD

**Discussion by MabezDev** [MabezDev/blog#10] [Ideas]
Drop some things you'd like to be discussed in the next blog post here!

**Comment by jessebraham** [comment]
Just doing a bit of a brain dump here, so feel free to filter as needed.

### xtensa-toolchain

- `xtensa-toolchain` GitHub Action now published on GitHub Marketplace
    - https://github.com/esp-rs/xtensa-toolchain
    - https://github.com/marketplace/actions/rust-toolchain-for-xtensa

### espflash

- Released version `1.3.0`:
    - Support multiple storage partitions and empty partition offsets, flashing to partitions other than `factory`
    - Allow for flashing via the USB Serial JTAG peripheral
    - Add the ability to pass unstable cargo features
- Upcoming in `1.4.0`:
    - Application firmware header can now have its values modified via command-line options
    - Use `espmonitor`
    - Various small refactors and bug fixes

### esp-hal

- Continued work on `esp-hal`
    - `GPIO`, `TIMG`, `UART`, and `Delay` working for ESP32/ESP32-C3
    - Initial support for interrupts added
    - ~~Support for ESP32-S2 and ESP32-S3 coming soon™️ ~~
    - See @bjoernQ's post for further updates

### esp-pacs

- Created `esp-pacs` ~~(to be transferred to `esp-rs` shortly)~~
    - https://github.com/esp-rs/esp-pacs
    - No longer depends on `make` and `python`

### miscellaneous

- Re-organization and more information added to book
    - Re-wrote section on `no_std` development to mention `esp-pacs` and `esp-hal`
- Note: we should mention our new development board
    - https://github.com/esp-rs/esp-rust-board

**Reply by jessebraham** [reply]
@MabezDev please ping be shortly before you plan on writing the post and I will update the above comment.

**Reply by MabezDev** [reply]
About to start this off :)

**Reply by jessebraham** [reply]
Updated my comment, probably still missing some things but I think I covered the big points.

**Comment by bjoernQ** [comment]
## esp-hal
- Support for ESP32-S2 and ESP32-S3 added
- Support for I2C, SPI and RNG added
